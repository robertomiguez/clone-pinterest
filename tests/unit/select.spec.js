import { shallowMount } from '@vue/test-utils'
import Select from '@/components/forms/Select.vue'

let wrapper

beforeEach(() => {
  wrapper = shallowMount(Select, {
    propsData: {
      options: ['op1', 'op2']
    },
    mocks: {},
    stubs: {},
    methods: {}
  })
})

afterEach(() => {
  wrapper.destroy()
})

describe('Card', () => {
  test('testing component props', () => {
    expect(wrapper.find('.select-css').text())
      .toBe(`Select one 
        op1
    
        op2`)
  })

  test('regression test', () => {
    expect(wrapper.element)
      .toMatchSnapshot()
  })
})
