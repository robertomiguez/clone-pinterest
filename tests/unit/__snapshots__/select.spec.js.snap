// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card regression test 1`] = `
<select
  class="select-css"
>
  <option
    value=""
  >
    Select one
  </option>
   
  <option
    value="op1"
  >
    
        op1
    
  </option>
  <option
    value="op2"
  >
    
        op2
    
  </option>
</select>
`;
